packages/ui/src/context/hooks/useBridgeChainHasNotEnoughNativeTokensToSend.ts

```diff
@@ -0,0 +1,80 @@
+import {
+    getAccountNativeTokenBalanceForChain,
+    fetchLatestGasPriceForChain,
+} from "../commActions"
+import { useMemo, useEffect } from "react"
+import useAsyncInvoke, { Status } from "../../util/hooks/useAsyncInvoke"
+import { BigNumber } from "ethers"
+import { GasPriceData } from "@block-wallet/background/controllers/GasPricesController"
+import { hasEnoughFundsToPayTheGasInSendTransaction } from "../../util/bridgeUtils"
+import { useBlankState } from "../../context/background/backgroundHooks"
+import { isNativeTokenAddress } from "../../util/tokenUtils"
+
+export enum EnoughNativeTokensToSend {
+    UNKNOWN = "UNKNOWN",
+    ENOUGH = "ENOUGH",
+    NOT_ENOUGH = "NOT_ENOUGH",
+}
+
+interface NativeAndGasPrices {
+    nativeTokenBalance: BigNumber | undefined
+    gasPrices: GasPriceData | undefined
+}
+
+export const useAddressHasEnoughNativeTokensToSend = (
```

Comment:
can we explain this hook in a comment? i.e.: Checks the native token balance on the destination network when performing a bridge involving a different token. Returns true if user has enough balance to perform a send tx